seek.go

```diff
@@ -74,12 +73,25 @@ func (s *SeekBuffer) Rest(cursor int) []byte {
 	return packet
 }
 
+// Resize removes redundant packets from the buffer by replacing the buffer's data
+// with a new array that is the remainder of the buffer after the cursor.
+func (s *SeekBuffer) Resize() {
+	s.dataLock.Lock()
+	defer s.dataLock.Unlock()
+
+	data := make([][]byte, len(s.data)-s.Cursor())
+	copy(data, s.data[s.Cursor():])
```

Comment:
Assumption that `len(s.data)>=s.Cursor()` always holds, yes? 

Not sure on the correctness of this, but benchmark in [2] is pretty interesting. 
[1] https://github.com/go101/go101/wiki/How-to-perfectly-clone-a-slice%3F
[2] https://github.com/go101/go101/wiki/How-to-efficiently-clone-a-slice%3F